### PR TITLE
Issue 414: use a clone of baseconf in new ServerConfiguration(baseConf)

### PR DESCRIPTION
**Pull request by zhaijack**
Descriptions of the changes in this PR:
This PR fix Issue #414 , and in the comments of that issue contains the anysis of the bug.  Main changes:
- Privde a clone of baseConf while do loadconf(baseConf), so make baseConf and newConf work independently.
- Remove 'loadConf(Configuration otherConf) ', since Configuration is an interface in common-configs and not have a clone method.

---
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

- [X] Make sure the PR title is formatted like:
    `<Issue # or BOOKKEEPER-#>: Description of pull request`
    `e.g. Issue 123: Description ...`
    `e.g. BOOKKEEPER-1234: Description ...`
- [ ] Make sure tests pass via `mvn clean apache-rat:check install findbugs:check`.
- [X] Replace `<Issue # or BOOKKEEPER-#>` in the title with the actual Issue/JIRA number.

---

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java [hunk 1]
```diff
@@ -127,18 +127,8 @@ public void loadConf(URL confURL) throws ConfigurationException {
...
-
-    /**
-     * Load configuration from other configuration object
-     *
-     * @param otherConf
-     *          Other configuration object
-     */
-    public void loadConf(Configuration otherConf) {
-        addConfiguration(otherConf);
+    public void loadConf(CompositeConfiguration baseConf) {
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java]: Any reason you changed this to `CompositeConfiguration`? You can do

```java
addConfiguration((Configuration) baseConf.clone());
```

Comment (zhaijack) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java]: As descripted above, configuration is an interface provided in common-configs, and not extends cloneable interface.  
This part of change is mainly to delete:
```
public void loadConf(Configuration otherConf) {
        addConfiguration(otherConf);
}
```
And change other place and compromise this method is calling.

Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java]: make sense.

**Comment by sijie** [comment]
@zhaijack fyi - I am testing a merge script change on updating labels, milestone and assignee. don't worry about the label changes here :)

**Pull request closed without merging**